In [1]:
import json


def load_level_data(level=0):
    with open(f"/home/paco/Documents/site_similarity/data/scrapping_results/processed level {level} results.json", 'r') as f:
        data = json.load(f)

    print(f'Loaded level {level} data with key size:{len(data)} values size:{sum([len(v) for v in data.values()])}')

    return data

In [2]:
lvl_zero_data = load_level_data()

Loaded level 0 data with key size:1062 values size:1062


In [3]:
lvl_one_data = load_level_data(level=1)

Loaded level 1 data with key size:1062 values size:4837


In [4]:
lvl_two_data = load_level_data(level=2)

Loaded level 2 data with key size:3560 values size:17408


In [5]:
lvl_zero_data['amazingnews.net']

{'amazingnews.net': {'site': 'amazingnews.net',
  'score': [],
  'audience_geography': []}}

In [6]:
list(lvl_zero_data.values())[0]

{'villagevoice.com': {'site': 'villagevoice.com',
  'score': [{'url': 'nylon.com', 'overlap_score': 4.1, 'alexa_rank': 51694.0},
   {'url': 'slantmagazine.com', 'overlap_score': 2.5, 'alexa_rank': 92418.0},
   {'url': 'gothamist.com', 'overlap_score': 2.4, 'alexa_rank': 16813.0},
   {'url': 'screendaily.com', 'overlap_score': 2.1, 'alexa_rank': 110772.0},
   {'url': 'amny.com', 'overlap_score': 2.1, 'alexa_rank': 69192.0}],
  'audience_geography': [{'country': 'United States', 'percent': 77.2}]}}

In [7]:
type(lvl_zero_data['villagevoice.com']), len(lvl_zero_data['villagevoice.com'])

(dict, 1)

In [8]:
type(lvl_one_data['villagevoice.com']), len(lvl_one_data['villagevoice.com'])

(dict, 5)

In [9]:
type(lvl_two_data['nylon.com']), len(lvl_two_data['nylon.com'])

(dict, 5)

In [10]:
from datetime import datetime
import os


def normalize_sites(lvl_data, result_data=[], level=0):

    sites = []
    overlap_sites = []
    audience_geography = []
    levels = []
    dates = []

    for big_k, v in lvl_data.items():
        for k in v:
            if not result_data or k not in result_data['sites'].values:
                if k == 'anullws.co':
                    print(big_k, v)
                sites.append(k)
                overlap_sites.append(v[k]['score'])
                audience_geography.append(v[k]['audience_geography'])
                levels.append(level)
                if os.path.exists(f"/home/paco/Documents/site_similarity/data/all_data/{k}.html"):
                    dates.append(datetime.fromtimestamp(os.stat(f"/home/paco/Documents/site_similarity/data/all_data/{k}.html").st_mtime))
                else:
                    dates.append(None)
    return [sites, overlap_sites, audience_geography, levels, dates]



import pandas as pd


total_data = normalize_sites(lvl_zero_data, [])
r0 = pd.DataFrame(total_data).transpose().rename(columns={0:'sites', 1:'overlap_sites', 2:'audience_geography', 3:'levels', 4:'dates'})

r0.head()

sites                                      overlap_sites  \
0       villagevoice.com  [{'url': 'nylon.com', 'overlap_score': 4.1, 'a...   
1  insideclimatenews.org  [{'url': 'wri.org', 'overlap_score': 12.8, 'al...   
2              fury.news  [{'url': 'settingrecordstraight.blogspot.com',...   
3           now8news.com  [{'url': 'donatebits.com', 'overlap_score': 11...   
4       constitution.com  [{'url': 'themilreview.com', 'overlap_score': ...   

                                  audience_geography levels  \
0    [{'country': 'United States', 'percent': 77.2}]      0   
1  [{'country': 'United States', 'percent': 69.0}...      0   
2                                                 []      0   
3                                                 []      0   
4                                                 []      0   

                       dates  
0 2020-06-29 19:55:15.620467  
1 2020-07-03 18:41:39.438563  
2 2020-06-29 19:55:17.684495  
3 2020-07-03 18:50:18.138572  
4 2020-07-03 18:53:18.803963

In [11]:
r0['dates'].min(), r0['dates'].max() 

(Timestamp('2020-06-29 19:55:15.620467'),
 Timestamp('2020-07-05 13:28:06.672250'))

In [12]:
'villagevoice.com' in r0['sites'].values

True

In [13]:
for k in list(lvl_one_data.values())[0]:
    print(k)
    #['villagevoice.com']

nylon.com
slantmagazine.com
gothamist.com
screendaily.com
amny.com


In [14]:
lvl_one_data['villagevoice.com']

{'nylon.com': {'site': 'nylon.com',
  'score': [{'url': 'compatible-astrology.com',
    'overlap_score': 16.1,
    'alexa_rank': 173807.0},
   {'url': 'astrology-zodiac-signs.com',
    'overlap_score': 14.1,
    'alexa_rank': 18913.0},
   {'url': 'papermag.com', 'overlap_score': 13.8, 'alexa_rank': 31762.0},
   {'url': 'astrology.com', 'overlap_score': 12.7, 'alexa_rank': 16859.0},
   {'url': 'flaunt.com', 'overlap_score': 11.1, 'alexa_rank': 10267.0}],
  'audience_geography': [{'country': 'United States', 'percent': 75.0},
   {'country': 'Canada', 'percent': 4.8},
   {'country': 'India', 'percent': 2.3}]},
 'slantmagazine.com': {'site': 'slantmagazine.com',
  'score': [{'url': 'criterion.com',
    'overlap_score': 11.3,
    'alexa_rank': 38257.0},
   {'url': 'rogerebert.com', 'overlap_score': 5.2, 'alexa_rank': 13105.0},
   {'url': 'empireonline.com', 'overlap_score': 4.6, 'alexa_rank': 37046.0},
   {'url': 'dvdbeaver.com', 'overlap_score': 4.6, 'alexa_rank': 291202.0},
   {'url': 'le

In [15]:
lvl_two_df = normalize_sites(lvl_one_data, [], level=1)


r1 = pd.DataFrame(lvl_two_df).transpose().rename(columns={0:'sites', 1:'overlap_sites', 2:'audience_geography', 3:'levels', 4:'dates'})

In [16]:
r1.head()

sites                                      overlap_sites  \
0          nylon.com  [{'url': 'compatible-astrology.com', 'overlap_...   
1  slantmagazine.com  [{'url': 'criterion.com', 'overlap_score': 11....   
2      gothamist.com  [{'url': 'nbcnewyork.com', 'overlap_score': 18...   
3    screendaily.com  [{'url': 'cineuropa.org', 'overlap_score': 15....   
4           amny.com  [{'url': 'gothamist.com', 'overlap_score': 15....   

                                  audience_geography levels  \
0  [{'country': 'United States', 'percent': 75.0}...      1   
1  [{'country': 'United States', 'percent': 54.8}...      1   
2  [{'country': 'United States', 'percent': 93.8}...      1   
3  [{'country': 'United States', 'percent': 19.6}...      1   
4  [{'country': 'United States', 'percent': 90.6}...      1   

                       dates  
0 2020-07-02 16:57:20.845460  
1 2020-07-02 16:57:32.397555  
2 2020-07-03 18:41:04.942296  
3 2020-07-02 16:57:54.349734  
4 2020-07-03 18:40:04.853829

In [17]:
r1.shape

(4837, 5)

In [18]:
'villagevoice.com' in r1['sites'].values

False

In [19]:
lvl_three_df = normalize_sites(lvl_two_data, [], level=2)


r2 = pd.DataFrame(lvl_three_df).transpose().rename(columns={0:'sites', 1:'overlap_sites', 2:'audience_geography', 3:'levels', 4:'dates'})

breaking-news.ca {'thewatchtowers.org': {'site': 'thewatchtowers.org', 'score': [{'url': 'breaking-news.ca', 'overlap_score': 10.3, 'alexa_rank': 8791216.0}, {'url': 'breakingwide.com', 'overlap_score': 7.4, 'alexa_rank': 376105.0}, {'url': 'kwpa.gov.ir', 'overlap_score': 3.9, 'alexa_rank': None}, {'url': 'praecursator007.co.uk', 'overlap_score': 3.8, 'alexa_rank': 3007841.0}, {'url': 'magamedia.org', 'overlap_score': 3.6, 'alexa_rank': 704833.0}], 'audience_geography': []}, 'pdgc2014c.wordpress.com': {'site': 'pdgc2014c.wordpress.com', 'score': [{'url': 'fabmimi.com', 'overlap_score': 7.0, 'alexa_rank': 1087711.0}, {'url': 'breaking-news.ca', 'overlap_score': 5.9, 'alexa_rank': 8791216.0}, {'url': 'fbinvestigations.blogspot.com', 'overlap_score': 2.9, 'alexa_rank': None}], 'audience_geography': []}, 'vivaliberty.com': {'site': 'vivaliberty.com', 'score': [{'url': 'truthcdm.com', 'overlap_score': 5.7, 'alexa_rank': None}, {'url': 'breaking-news.ca', 'overlap_score': 4.9, 'alexa_rank': 

In [20]:

r2.head()

sites  \
0    compatible-astrology.com   
1  astrology-zodiac-signs.com   
2                papermag.com   
3               astrology.com   
4                  flaunt.com   

                                       overlap_sites  \
0  [{'url': 'astrology-zodiac-signs.com', 'overla...   
1  [{'url': 'astrology.com', 'overlap_score': 69....   
2  [{'url': 'nylon.com', 'overlap_score': 13.8, '...   
3  [{'url': 'astrology-zodiac-signs.com', 'overla...   
4  [{'url': 'earmilk.com', 'overlap_score': 19.6,...   

                                  audience_geography levels  \
0  [{'country': 'United States', 'percent': 41.7}...      2   
1  [{'country': 'United States', 'percent': 46.2}...      2   
2  [{'country': 'United States', 'percent': 81.7}...      2   
3  [{'country': 'United States', 'percent': 44.8}...      2   
4  [{'country': 'United States', 'percent': 58.36...      2   

                       dates  
0 2020-07-03 14:14:57.211172  
1 2020-07-03 14:15:08.879345  
2 2020-07-03 14:15:20.427515  
3 2020-07-03 14:15:31.859682  
4 2020-07-03 14:15:43.379848

In [21]:
r2[r2['dates'].isnull()]

sites                                      overlap_sites  \
5424            anullws.co  [{'url': 'weloveanimals.me', 'overlap_score': ...   
13452  nullsuchrecipes.com  [{'url': 'suicidefood.blogspot.com', 'overlap_...   

      audience_geography levels dates  
5424                  []      2   NaT  
13452                 []      2   NaT

In [22]:

lvl_three_data = load_level_data(level=3)

Loaded level 3 data with key size:10650 values size:51865


In [23]:
lvl_four_df = normalize_sites(lvl_three_data, [], level=3)


r3 = pd.DataFrame(lvl_four_df).transpose().rename(columns={0:'sites', 1:'overlap_sites', 2:'audience_geography', 3:'levels', 4:'dates'})

In [24]:
r3.head()

sites  \
0  astrology-zodiac-signs.com   
1               astrology.com   
2             astromatcha.com   
3           ganeshaspeaks.com   
4              astrostyle.com   

                                       overlap_sites  \
0  [{'url': 'astrology.com', 'overlap_score': 69....   
1  [{'url': 'astrology-zodiac-signs.com', 'overla...   
2  [{'url': 'compatible-astrology.com', 'overlap_...   
3  [{'url': 'astrology.com', 'overlap_score': 49....   
4  [{'url': 'astrology.com', 'overlap_score': 59....   

                                  audience_geography levels  \
0  [{'country': 'United States', 'percent': 46.2}...      3   
1  [{'country': 'United States', 'percent': 44.8}...      3   
2                                                 []      3   
3  [{'country': 'India', 'percent': 68.2}, {'coun...      3   
4  [{'country': 'United States', 'percent': 53.2}...      3   

                       dates  
0 2020-07-03 14:15:08.879345  
1 2020-07-03 14:15:31.859682  
2 2020-07-08 16:29:55.575374  
3 2020-07-08 16:30:06.687489  
4 2020-07-08 16:30:17.867604

In [25]:

r3.shape

(51865, 5)

In [26]:
r3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51865 entries, 0 to 51864
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   sites               51865 non-null  object        
 1   overlap_sites       51865 non-null  object        
 2   audience_geography  51865 non-null  object        
 3   levels              51865 non-null  object        
 4   dates               51865 non-null  datetime64[ns]
dtypes: datetime64[ns](1), object(4)
memory usage: 2.0+ MB


In [27]:

r3[r3['dates'].isnull()]

Empty DataFrame
Columns: [sites, overlap_sites, audience_geography, levels, dates]
Index: []

In [29]:
lvl_four_data = load_level_data(level=4)


Loaded level 4 data with key size:28871 values size:139994


In [30]:

lvl_fifth_df = normalize_sites(lvl_four_data, [], level=4)


r4 = pd.DataFrame(lvl_fifth_df).transpose().rename(columns={0:'sites', 1:'overlap_sites', 2:'audience_geography', 3:'levels', 4:'dates'})

In [31]:
r4.head()

sites  \
0             astrology.com   
1  compatible-astrology.com   
2            astrostyle.com   
3           thehoroscope.co   
4         ganeshaspeaks.com   

                                       overlap_sites  \
0  [{'url': 'astrology-zodiac-signs.com', 'overla...   
1  [{'url': 'astrology-zodiac-signs.com', 'overla...   
2  [{'url': 'astrology.com', 'overlap_score': 59....   
3  [{'url': 'sunsigns.org', 'overlap_score': 55.3...   
4  [{'url': 'astrology.com', 'overlap_score': 49....   

                                  audience_geography levels  \
0  [{'country': 'United States', 'percent': 44.8}...      4   
1  [{'country': 'United States', 'percent': 41.7}...      4   
2  [{'country': 'United States', 'percent': 53.2}...      4   
3  [{'country': 'United States', 'percent': 49.3}...      4   
4  [{'country': 'India', 'percent': 68.2}, {'coun...      4   

                       dates  
0 2020-07-03 14:15:31.859682  
1 2020-07-03 14:14:57.211172  
2 2020-07-08 16:30:17.867604  
3 2020-07-08 16:30:30.499735  
4 2020-07-08 16:30:06.687489

In [33]:
r4.shape

(139994, 5)

In [32]:

r4[r4['dates'].isnull()]


Empty DataFrame
Columns: [sites, overlap_sites, audience_geography, levels, dates]
Index: []

In [34]:
total_df = pd.concat([r0, r1, r2, r3, r4])

In [35]:
total_df.shape

(215166, 5)

In [36]:

total_df.head()

sites                                      overlap_sites  \
0       villagevoice.com  [{'url': 'nylon.com', 'overlap_score': 4.1, 'a...   
1  insideclimatenews.org  [{'url': 'wri.org', 'overlap_score': 12.8, 'al...   
2              fury.news  [{'url': 'settingrecordstraight.blogspot.com',...   
3           now8news.com  [{'url': 'donatebits.com', 'overlap_score': 11...   
4       constitution.com  [{'url': 'themilreview.com', 'overlap_score': ...   

                                  audience_geography levels  \
0    [{'country': 'United States', 'percent': 77.2}]      0   
1  [{'country': 'United States', 'percent': 69.0}...      0   
2                                                 []      0   
3                                                 []      0   
4                                                 []      0   

                       dates  
0 2020-06-29 19:55:15.620467  
1 2020-07-03 18:41:39.438563  
2 2020-06-29 19:55:17.684495  
3 2020-07-03 18:50:18.138572  
4 2020-07-03 18:53:18.803963

In [37]:
len(total_df['sites'].unique())

78431

In [52]:
list(set(r3['sites'].values.tolist()).difference(set(r4['sites'].values.tolist())))[:10]

['eastasiaforum.org',
 'thewatchmakerproject.com',
 'barbarycoasttrail.org',
 'ewanpaterson.net',
 'theraisedright.com',
 'massterlist.com',
 'streetheattv.com',
 'grantcountywa.gov',
 'essexanimalhospital.com',
 'cienciologia.wordpress.com']

In [53]:
total_df[total_df['sites'] == 'eastasiaforum.org']

sites                                      overlap_sites  \
340   eastasiaforum.org  [{'url': 'thediplomat.com', 'overlap_score': 6...   
5723  eastasiaforum.org  [{'url': 'thediplomat.com', 'overlap_score': 6...   

                                     audience_geography levels  \
340   [{'country': 'Cambodia', 'percent': 4.9}, {'co...      1   
5723  [{'country': 'Cambodia', 'percent': 4.9}, {'co...      3   

                          dates  
340  2020-07-02 18:00:03.220459  
5723 2020-07-02 18:00:03.220459

In [58]:
total_df[total_df.apply(lambda x: any('eastasiaforum.org' in y['url'] for y in  x['overlap_sites']), axis=1)]

sites                                      overlap_sites  \
78              atimes.com  [{'url': 'thediplomat.com', 'overlap_score': 3...   
1627    eastwestcenter.org  [{'url': 'eastasiaforum.org', 'overlap_score':...   
16480   eastwestcenter.org  [{'url': 'eastasiaforum.org', 'overlap_score':...   
115699             nbr.org  [{'url': 'spfusa.org', 'overlap_score': 2.1, '...   
115732             nbr.org  [{'url': 'spfusa.org', 'overlap_score': 2.1, '...   

       audience_geography levels                      dates  
78                     []      0 2020-06-29 19:56:15.497266  
1627                   []      2 2020-07-03 22:19:44.514778  
16480                  []      4 2020-07-03 22:19:44.514778  
115699                 []      4 2020-07-18 11:42:39.919250  
115732                 []      4 2020-07-18 11:42:39.919250

In [45]:
total_df.to_csv(f"/home/paco/Documents/site_similarity/data/total_data.csv", index=False)

In [46]:
total_df[total_df['dates'].isnull()]

sites                                      overlap_sites  \
5424            anullws.co  [{'url': 'weloveanimals.me', 'overlap_score': ...   
13452  nullsuchrecipes.com  [{'url': 'suicidefood.blogspot.com', 'overlap_...   

      audience_geography levels dates  
5424                  []      2   NaT  
13452                 []      2   NaT

In [49]:
total_df.apply(lambda x: len(x['overlap_sites']), axis=1).value_counts()

5    205697
3      2778
4      2743
2      2327
1      1448
0       173
dtype: int64

In [62]:
unique_df = total_df.drop_duplicates(subset=['sites'])

In [63]:
unique_df.shape

(78431, 5)

In [67]:
unique_df.groupby('levels').count()

sites  overlap_sites  audience_geography  dates
levels                                                 
0        1062           1062                1062   1062
1        3176           3176                3176   3176
2        7631           7631                7631   7629
3       19022          19022               19022  19022
4       47540          47540               47540  47540

Loaded level 0 key size:1062  values size:1062  
Loaded level 1 key size:1062  values size:4837  
Loaded level 2 key size:3560  values size:17408  
Loaded level 3 key size:10650 values size:51865  
Loaded level 4 key size:28871 values size:139994  

In [68]:
unique_df[unique_df.apply(lambda x: any('eastasiaforum.org' in y['url'] for y in  x['overlap_sites']), axis=1)]

sites                                      overlap_sites  \
78              atimes.com  [{'url': 'thediplomat.com', 'overlap_score': 3...   
1627    eastwestcenter.org  [{'url': 'eastasiaforum.org', 'overlap_score':...   
115699             nbr.org  [{'url': 'spfusa.org', 'overlap_score': 2.1, '...   

       audience_geography levels                      dates  
78                     []      0 2020-06-29 19:56:15.497266  
1627                   []      2 2020-07-03 22:19:44.514778  
115699                 []      4 2020-07-18 11:42:39.919250

In [69]:
unique_df[unique_df['sites'] == 'eastasiaforum.org']

sites                                      overlap_sites  \
340  eastasiaforum.org  [{'url': 'thediplomat.com', 'overlap_score': 6...   

                                    audience_geography levels  \
340  [{'country': 'Cambodia', 'percent': 4.9}, {'co...      1   

                         dates  
340 2020-07-02 18:00:03.220459

In [70]:
unique_df[unique_df['dates'].isnull()]

sites                                      overlap_sites  \
5424            anullws.co  [{'url': 'weloveanimals.me', 'overlap_score': ...   
13452  nullsuchrecipes.com  [{'url': 'suicidefood.blogspot.com', 'overlap_...   

      audience_geography levels dates  
5424                  []      2   NaT  
13452                 []      2   NaT

In [73]:
unique_df[unique_df.apply(lambda x: any('anullws.co' in y['url'] for y in  x['overlap_sites']), axis=1)]

Empty DataFrame
Columns: [sites, overlap_sites, audience_geography, levels, dates]
Index: []

In [74]:
unique_df[unique_df.apply(lambda x: any('nullsuchrecipes.com' in y['url'] for y in  x['overlap_sites']), axis=1)]

Empty DataFrame
Columns: [sites, overlap_sites, audience_geography, levels, dates]
Index: []

In [83]:
from bs4 import BeautifulSoup


def get_referal_sites(site):
    file_name = f"/home/paco/Documents/site_similarity/data/all_data/{site}.html"

    if not os.path.exists(file_name):
        return
    
    with open(file_name) as f:
        text = f.read()

    text = BeautifulSoup(text, 'html')

    referal_sites = text.find('div', {'id': 'card_referralsites'})

    if not referal_sites:
        return []

    found_sites = referal_sites.find_all('div', {'class': 'Row'})

    return [(found_site.find('div', {'class', 'site'}).a['href'].split('/')[-1], 
            found_site.find('span', {'class': 'truncation'}).text.strip())
            for found_site in found_sites]

In [84]:
get_referal_sites('eastasiaforum.org')


[('atimes.com', '8 k'),
 ('thediplomat.com', '3.6 k'),
 ('eastasiaforum.org', '1.1 k'),
 ('lowyinstitute.org', '570'),
 ('rsis.edu.sg', '407')]

In [85]:
unique_df['referal_sites'] = unique_df['sites'].apply(get_referal_sites)

In [86]:
unique_df.to_csv(f"/home/paco/Documents/site_similarity/data/unique_data.csv", index=False)

In [87]:
unique_df.head()

sites                                      overlap_sites  \
0       villagevoice.com  [{'url': 'nylon.com', 'overlap_score': 4.1, 'a...   
1  insideclimatenews.org  [{'url': 'wri.org', 'overlap_score': 12.8, 'al...   
2              fury.news  [{'url': 'settingrecordstraight.blogspot.com',...   
3           now8news.com  [{'url': 'donatebits.com', 'overlap_score': 11...   
4       constitution.com  [{'url': 'themilreview.com', 'overlap_score': ...   

                                  audience_geography levels  \
0    [{'country': 'United States', 'percent': 77.2}]      0   
1  [{'country': 'United States', 'percent': 69.0}...      0   
2                                                 []      0   
3                                                 []      0   
4                                                 []      0   

                       dates  \
0 2020-06-29 19:55:15.620467   
1 2020-07-03 18:41:39.438563   
2 2020-06-29 19:55:17.684495   
3 2020-07-03 18:50:18.138572   
4 2020-07-03 18:53:18.803963   

                                       referal_sites  
0  [(villagevoice.com, 13.1 k), (gothamist.com, 8...  
1  [(climatecentral.org, 2.3 k), (skepticalscienc...  
2                                                 []  
3  [(now8news.com, 90), (notallowedto.com, 53), (...  
4  [(zfacts.com, 1.1 k), (eaglerising.com, 273), ...

In [96]:
unique_df.shape

(78431, 6)

In [97]:
cleaned_df = unique_df[unique_df['referal_sites'].notnull()]

In [98]:
test_df = pd.read_csv('/home/paco/Documents/site_similarity/data/unique_data.csv')

In [99]:
test_df.head()

sites                                      overlap_sites  \
0       villagevoice.com  [{'url': 'nylon.com', 'overlap_score': 4.1, 'a...   
1  insideclimatenews.org  [{'url': 'wri.org', 'overlap_score': 12.8, 'al...   
2              fury.news  [{'url': 'settingrecordstraight.blogspot.com',...   
3           now8news.com  [{'url': 'donatebits.com', 'overlap_score': 11...   
4       constitution.com  [{'url': 'themilreview.com', 'overlap_score': ...   

                                  audience_geography  levels  \
0    [{'country': 'United States', 'percent': 77.2}]       0   
1  [{'country': 'United States', 'percent': 69.0}...       0   
2                                                 []       0   
3                                                 []       0   
4                                                 []       0   

                        dates  \
0  2020-06-29 19:55:15.620467   
1  2020-07-03 18:41:39.438563   
2  2020-06-29 19:55:17.684495   
3  2020-07-03 18:50:18.138572   
4  2020-07-03 18:53:18.803963   

                                       referal_sites  
0  [('villagevoice.com', '13.1 k'), ('gothamist.c...  
1  [('climatecentral.org', '2.3 k'), ('skepticals...  
2                                                 []  
3  [('now8news.com', '90'), ('notallowedto.com', ...  
4  [('zfacts.com', '1.1 k'), ('eaglerising.com', ...

In [109]:
test_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78431 entries, 0 to 78430
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   sites               78431 non-null  object
 1   overlap_sites       78431 non-null  object
 2   audience_geography  78431 non-null  object
 3   levels              78431 non-null  int64 
 4   dates               78429 non-null  object
 5   referal_sites       78429 non-null  object
dtypes: int64(1), object(5)
memory usage: 3.6+ MB


In [113]:
v = json.dumps(unique_df.head()['overlap_sites'][0])

In [114]:
json.loads(v)

[{'url': 'nylon.com', 'overlap_score': 4.1, 'alexa_rank': 51694.0},
 {'url': 'slantmagazine.com', 'overlap_score': 2.5, 'alexa_rank': 92418.0},
 {'url': 'gothamist.com', 'overlap_score': 2.4, 'alexa_rank': 16813.0},
 {'url': 'screendaily.com', 'overlap_score': 2.1, 'alexa_rank': 110772.0},
 {'url': 'amny.com', 'overlap_score': 2.1, 'alexa_rank': 69192.0}]

In [125]:
import numpy as np


unique_df.head()['dates'].astype(np.int64) // 10**9

0    1593460515
1    1593801699
2    1593460517
3    1593802218
4    1593802398
Name: dates, dtype: int64

In [129]:
unique_df.head()['dates']

0   2020-06-29 19:55:15.620467
1   2020-07-03 18:41:39.438563
2   2020-06-29 19:55:17.684495
3   2020-07-03 18:50:18.138572
4   2020-07-03 18:53:18.803963
Name: dates, dtype: datetime64[ns]

In [128]:
from datetime import datetime

pd.Timestamp(1593802398, unit='s')

Timestamp('2020-07-03 18:53:18')

In [135]:
cleaned_df['created_timestamp'] = cleaned_df['dates'].astype(np.int64) // 10**9

In [136]:
cleaned_df.columns

Index(['sites', 'overlap_sites', 'audience_geography', 'levels', 'dates',
       'referal_sites', 'created_timestamp'],
      dtype='object')

In [147]:
with open('clean_data_20200803.json', 'w') as f:
    json.dump(cleaned_df[['sites', 'overlap_sites', 'referal_sites', 'audience_geography', 'levels', 'created_timestamp']].to_dict('records'), f, indent=4)

In [148]:
with open('clean_data_20200803.json', 'r') as f:
    data = json.load(f)

In [149]:
len(data)

78429

In [133]:
os.stat(f"/home/paco/Documents/site_similarity/data/all_data/villagevoice.com.html").st_mtime

1593449715.620467

In [ ]:
test = {'thewatchtowers.org': {'site': 'thewatchtowers.org', 'score': [{'url': 'breaking-news.ca', 'overlap_score': 10.3, 'alexa_rank': 8791216.0}, {'url': 'breakingwide.com', 'overlap_score': 7.4, 'alexa_rank': 376105.0}, {'url': 'kwpa.gov.ir', 'overlap_score': 3.9, 'alexa_rank': None}, {'url': 'praecursator007.co.uk', 'overlap_score': 3.8, 'alexa_rank': 3007841.0}, {'url': 'magamedia.org', 'overlap_score': 3.6, 'alexa_rank': 704833.0}], 'audience_geography': []}, 'pdgc2014c.wordpress.com': {'site': 'pdgc2014c.wordpress.com', 'score': [{'url': 'fabmimi.com', 'overlap_score': 7.0, 'alexa_rank': 1087711.0}, {'url': 'breaking-news.ca', 'overlap_score': 5.9, 'alexa_rank': 8791216.0}, {'url': 'fbinvestigations.blogspot.com', 'overlap_score': 2.9, 'alexa_rank': None}], 'audience_geography': []}, 'vivaliberty.com': {'site': 'vivaliberty.com', 'score': [{'url': 'truthcdm.com', 'overlap_score': 5.7, 'alexa_rank': None}, {'url': 'breaking-news.ca', 'overlap_score': 4.9, 'alexa_rank': 8791216.0}, {'url': 'redrightvideos.com', 'overlap_score': 1.9, 'alexa_rank': None}, {'url': 'reconsider.news', 'overlap_score': 1.5, 'alexa_rank': 10132041.0}, {'url': 'silenceisconsent.net', 'overlap_score': 1.1, 'alexa_rank': 9911326.0}], 'audience_geography': []}, 'fabmimi.com': {'site': 'fabmimi.com', 'score': [{'url': 'linknaija.com', 'overlap_score': 7.8, 'alexa_rank': 1158136.0}, {'url': 'pdgc2014c.wordpress.com', 'overlap_score': 7.0, 'alexa_rank': None}, {'url': 'atiterkula.wordpress.com', 'overlap_score': 5.0, 'alexa_rank': 4979546.0}, {'url': 'samuelortom.com', 'overlap_score': 4.5, 'alexa_rank': 2510639.0}, {'url': 'nigerianconcord.com.ng', 'overlap_score': 4.3, 'alexa_rank': None}], 'audience_geography': []}, 'anullws.co': {'site': 'anullws.co', 'score': [{'url': 'weloveanimals.me', 'overlap_score': 4.9, 'alexa_rank': 525321.0}, {'url': 'saydorsblog.com', 'overlap_score': 4.0, 'alexa_rank': None}, {'url': 'thedailynet.com', 'overlap_score': 2.5, 'alexa_rank': 4664291.0}, {'url': 'bentrovatowhippingboy.wordpress.com', 'overlap_score': 2.5, 'alexa_rank': None}, {'url': 'upliftingtoday.com', 'overlap_score': 2.2, 'alexa_rank': 264912.0}], 'audience_geography': []}}

In [42]:
test.keys()

dict_keys(['thewatchtowers.org', 'pdgc2014c.wordpress.com', 'vivaliberty.com', 'fabmimi.com', 'anullws.co'])